src/components/UniversalPageLayout.tsx

```diff
@@ -0,0 +1,375 @@
+/** @jsx jsx */
+import { LogOnMount } from '@amplitude/react-amplitude';
+import { css, Global, jsx } from '@emotion/core';
+import * as Sentry from '@sentry/node';
+import { isBrowser } from '@unly/utils';
+import { createLogger } from '@unly/utils-simple-logger';
+import classnames from 'classnames';
+import { ThemeProvider } from 'emotion-theming';
+import React from 'react';
+import { Button } from 'reactstrap';
+
+import { NRN_DEFAULT_FONT, NRN_DEFAULT_SECONDARY_COLOR } from '../constants';
+import ErrorPage from '../pages/_error';
+import { LayoutPageProps } from '../types/LayoutPageProps';
+import { getValue, STRATEGY_DO_NOTHING } from '../utils/record';
+import Footer from './Footer';
+import Head from './Head';
+import Nav from './Nav';
+
+const fileLabel = 'components/UniversalPageLayout';
+const logger = createLogger({ // eslint-disable-line no-unused-vars,@typescript-eslint/no-unused-vars
+  label: fileLabel,
+});
+
+type Props = {
+  children: Function;
+} & LayoutPageProps;
+
+/**
+ *
+ * @param props
+ * @constructor
+ */
+const UniversalPageLayout = (props: Props): JSX.Element => {
+  const {
+    cookiesManager,
+    customer,
+    customerRef,
+    defaultLocales,
+    error,
+    i18nextInstance,
+    iframeReferrer,
+    isInIframe,
+    headProps,
+    lang,
+    locale,
+    pageName,
+    router,
+    userSession,
+    theme,
+  } = props;
+  const { children, ...layoutPageProps } = props; // Only keep LayoutPageProps variables (remove children)
+
+  Sentry.addBreadcrumb({ // See https://docs.sentry.io/enriching-error-data/breadcrumbs
+    category: fileLabel,
+    message: `Rendering ${fileLabel} for page "${pageName}"`,
+    level: Sentry.Severity.Debug,
+  });
+
+  if (isBrowser()) { // Avoids log clutter on server
+    console.debug(`UniversalPageLayout.layoutPageProps`, layoutPageProps);
+  }
+
+  return (
+    <>
+      <Head {...headProps} />
+      <LogOnMount eventType="page-displayed" />
+
+      {/* XXX Global styles that applies to all pages within this layout go there */}
+      <Global
+        styles={css`
```

Comment:
This is a personal preference thing, but with big css content like this I feel that using the `@emotion/styled` style approach keeping styles perhaps in a different file would make things more readable.

Reply:
Agreed, if you have an example for global styles using a cleaner approach, I'm all ears! :)

Reply:
I've refactored it through https://github.com/UnlyEd/next-right-now/pull/42/commits/427f93616db8ad8b52fca41c0d58c51f7986e417

Please let me know what you think, I do believe it's cleaner and easier to maintain.

Reply:
That looks a lot better, there is also this approach just fyi that is super helpful:
```typescript
import styled from '@emotion/styled'

const Button = styled.button`
  // add css here
`
```
or
```typescript
const MyStyledComponent = styled(BasicComponent)`
  color: blue;
`
```

Reply:
Yup, agreed. It just doesn't fit here IMHO, but great for components.